### PR TITLE
feat: Add mergeParams utility function

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

File: src/metamorphoses.coffee [hunk 1]
```diff
@@ -39,6 +39,17 @@ createAdapter = (mimeType) ->
     return
 
 
+mergeParams = (resourceParams, actionParams) ->
```
Comment (kuba-kubula) [inline src/metamorphoses.coffee:42]: please provide default value of `[]` for both arguments.

**Comment by w-vi** [comment]
Will it be used somewhere? Is it an effort to de-duplicate code in core app?

**Comment by pksunkara** [comment]
Yes, this will be used in core app and we wanted it here so that it can be single source of truth.

**Comment by pksunkara** [comment]
https://github.com/apiaryio/apiary/commit/cf25fdf0482602247f839ae8add9e5d95a623635

**Review by kuba-kubula — COMMENTED** [review]
changes needed

**Comment by kuba-kubula** [comment]
Also missing test for these scenarios:
- 0 resource-params, 1 action param
- 1 resource-param, 0 action params
- overwriting multiple resource-params by multiple action-params with the same keys (multi-to-multi situation)

**Comment by pksunkara** [comment]
@kuba-kubula Here's the PR for those changes. https://github.com/apiaryio/metamorphoses/pull/61

**Pull request merged**